internal/cloud/realm/client.go

```diff
@@ -35,12 +58,36 @@ func (c *client) do(method, path string, options api.RequestOptions) (*http.Resp
 		return nil, err
 	}
 
-	req.Header = options.Header
-	if req.Header == nil {
-		req.Header = http.Header{}
-	}
+	req.Header = http.Header{}
```

Comment:
I don't follow what this is doing, req.Header should already be created. Are we recreating it for a specific purpose?

Reply:
I put this here, carryover from the previous codebase too.  my understanding was that since `http.Header` was a type alias of `map`, we needed to worry about `nil` maps before calling `Set`.  a few quick tests seems to confirm that is not the case...

Reply:
I just tested this out too, I will remove this